### PR TITLE
minor: revert parsing precedence between Aggr and UDAF

### PR DESCRIPTION
**Pull request by waynexia**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Just came across these lines that UDF is parsed before scalar function:

https://github.com/apache/arrow-datafusion/blob/bcc2acd8d10204b32b64fa3b77372655eaee8f06/datafusion/sql/src/expr/function.rs#L49-L61 

And then find this rule doesn't hold for Aggr/UDAF

## What changes are included in this PR?

After this patch, UDAF can shadow builtin AggrFn if they have the same name. 

## Are these changes tested?

n/a

## Are there any user-facing changes?

It is... I think

### CODE REVIEW

**Review by ozankabak — APPROVED** [review]
Makes sense, thank you

**Comment by alamb** [comment]
I agree it makes sense to allow UDAF to shadow built in aggregate functions (so that users can redefine the meaning of aggregates if they wanted)

Is there any way we can add a test to document this behavior? Perhaps in https://github.com/apache/arrow-datafusion/blob/3f8c51222fa0aa5bb4ef7fa5cab55875bfde0a63/datafusion/core/tests/user_defined/user_defined_aggregates.rs 🤔

**Comment by waynexia** [comment]
Thanks for reviewing!

>Is there any way we can add a test to document this behavior

Good suggestion 👍 I've added one in that file, reusing the existing `time_sum` to shadow the builtin `sum`, please take a look

File: datafusion/core/tests/user_defined/user_defined_aggregates.rs [hunk 1]
```diff
@@ -169,6 +169,37 @@ async fn test_udaf_returning_struct_subquery() {
...
+    let expected = [
+        "+-------------+",
+        "| SUM(t.time) |",
+        "+-------------+",
+        "| 19000       |",
+        "+-------------+",
+    ];
+    assert_batches_eq!(expected, &execute(&ctx, sql).await.unwrap());
+
+    // Register `TimeSum` with name `sum`. This will shadow the builtin one
```
Comment (alamb) [inline datafusion/core/tests/user_defined/user_defined_aggregates.rs:190]: 💯

**Review by alamb — APPROVED** [review]
Thanks @waynexia

**Pull request merged**